### PR TITLE
docs: document dep usage for developers

### PR DESCRIPTION
**Pull request by aanm**
Fixes https://github.com/cilium/cilium/issues/1972

Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4311)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: Documentation/contributing.rst [hunk 1]
```diff
@@ -901,6 +901,41 @@ for debugging what is going on inside them, for example:
...
+     git hash    : 37d9ea0a
+     go version  : go1.9.1
+     go compiler : gc
+     platform    : linux/amd64
+
+After that, you can edit the ``Gopkg.toml`` file, add the library that you want
+to add. Lets assume we want to add ``github.com/containernetworking/cni``
+version ``v0.5.2``:
+
+.. code:: bash
```
Comment (ianvernon) [inline Documentation/contributing.rst:924]: Nit: is this really bash code?

**Comment by ianvernon** [comment]
Build failed due to spellchecker failing due to golang-specific words added in this PR:
```
16:38:12 Spelling checker messages written to /srv/Documentation/_build/spelling/output.txt
16:38:12 WARNING: Found 3 misspelled words
16:38:12 build succeeded, 1 warning.
16:38:12 WARNING: Not copying tabs assets! Not compatible with spelling builder
16:38:12 Please fix the following spelling mistakes:
16:38:12 * Documentation/contributing.rst:905: (golang) 
16:38:12 * Documentation/contributing.rst:905: (dep) 
16:38:12 * Documentation/contributing.rst:907: (dep) 
16:38:12 If the words are not misspelled, add them to Documentation/spelling_wordlist.txt.
```

Please fix the spellchecker to ignore these words, and I will re-review.

File: Documentation/contributing.rst (hunk 1 above)
Comment (aanm) [reply Documentation/contributing.rst:924]: No, but there's no "toml" integration in docs

**Comment by aanm** [comment]
test-docs-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**